dependencies/Indy10/System/IdStackUnix.pas

```diff
@@ -75,20 +75,53 @@ interface
 {$ENDIF}
 uses
   Classes,
+  Contnrs,
   sockets,
   baseunix,
   IdStack,
   IdStackConsts,
   IdGlobal,
   IdStackBSDBase;
 
+
 {$IFDEF FREEBSD}
   {$DEFINE SOCK_HAS_SINLEN}
 {$ENDIF}
 {$IFDEF DARWIN}
   {$DEFINE SOCK_HAS_SINLEN}
 {$ENDIF}
 
+
+{$IFDEF CPUAARCH64}
+  {$IFDEF DARWIN}
+    {$DEFINE DNSCACHE}
+    // On Mac M1, it seems as though DNS resolution is not cached by the system.
+    // so we're going to cache it here.
+  {$ENDIF}
+{$ENDIF}
+

```

Comment:
I would suggest moving {$DEFINE DNSCACHE} higher up in the file, so that the inclusion of the Contnrs unit in the uses clause can then be wrapped in an {$IFDEF}.